doc/Cifar10.rst

```diff
@@ -0,0 +1,1124 @@
+            ) for i, _ in enumerate(alphas)
+        ]
+        sizes[name] = [
+            y_pss[name][:, :, i].sum(axis=1).mean() for i, _ in enumerate(alphas)
+        ]
+
+
+.. code-block:: python
+
+    coverage_90 = {method: coverage[9] for method, coverage in coverages.items()}
+    null_90 = {method: null[9] for method, null in nulls.items()}
+    width_90 = {method: width[9] for method, width in sizes.items()}
+    y_ps_90 = {method: y_ps[:, :, 9] for method, y_ps in y_pss.items()}
+
+Let’s now look at the marginal coverages, number of null prediction
+sets, and the averaged size of prediction sets for a confidence level of
+90 %.
+
+.. code-block:: python
+
+    summary_df = pd.concat(
+        [
+            pd.Series(coverage_90),
+            pd.Series(null_90),
+            pd.Series(width_90)
+        ],
+        axis=1,
+        keys=["Coverages", "Number of null sets", "Average prediction set sizes"]
+    ).round(3)
+
+.. code-block:: python
+
+    summary_df
+
+
+
+
+.. raw:: html
+
+    <div>
+    <style scoped>
+        .dataframe tbody tr th:only-of-type {
+            vertical-align: middle;
+        }
+    
+        .dataframe tbody tr th {
+            vertical-align: top;
+        }
+    
+        .dataframe thead th {
+            text-align: right;
+        }
+    </style>
+    <table border="1" class="dataframe">
+      <thead>
+        <tr style="text-align: right;">
+          <th></th>
+          <th>Coverages</th>
+          <th>Number of null sets</th>
+          <th>Average prediction set sizes</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <th>naive</th>
+          <td>0.732</td>
+          <td>0</td>
+          <td>1.258</td>
+        </tr>
+        <tr>
+          <th>score</th>
+          <td>0.912</td>
+          <td>0</td>
+          <td>2.356</td>
+        </tr>
+        <tr>
+          <th>cumulated_score</th>
+          <td>0.928</td>
+          <td>0</td>
+          <td>2.701</td>
+        </tr>
+        <tr>
+          <th>random_cumulated_score</th>
+          <td>0.908</td>
+          <td>21</td>
+          <td>2.463</td>
+        </tr>
+        <tr>
+          <th>top_k</th>
+          <td>0.910</td>
+          <td>0</td>
+          <td>3.000</td>
+        </tr>
+      </tbody>
+    </table>
+    </div>
+
+
+
+As expected, the “naive” method, which directly uses the alpha value as
+a threshold for selecting the prediction sets, does not give guarantees
+on the marginal coverage since this method is not calibrated. Other
+methods give a marginal coverage close to the desired one, i.e. 90%.
+Notice that the “cumulated_score” method, which always includes the last
+label whose cumulated score is above the given quantile, tends to give
+slightly higher marginal coverages since the prediction sets are
+slightly too big.
+
+4. Visualization of the prediction sets
+---------------------------------------
+
+Thanks to this really awesome function, you can generate an amazing
```

Comment:
Ahah, this sentence should be removed now :-)

Reply:
done